### ISSUE TITLE
Paypal Express Checkout always creates order as guest user

### ISSUE THREAD

**Issue by mfritsche-dotsource** [PAYONE-GmbH/shopware-5#150]
We have noticed a change in Paypal Express behaviour in the recent version 3.7.0. 

Before this release a check was made if the user is logged in and if not a new user was created. Code example Controllers/Frontend/MoptPaymentEcs.php createrOrUpdateAndForwardUser() function:
```
        if (!$this->isUserLoggedIn($session)) {
            $this->createUserWithoutAccount($payData, $session, $paymentId);
        } else {
            $user = $this->updateUserAddresses($payData, $session, $paymentId);
            if ($user === null) {
                return $this->ecsAbortAction();
            }
        }
```
This changed with 3.7.0. Now always a new user is created which also results in a new session and the user is not logged in anymore after using Paypal Express as payment mean. Code example Controllers/Frontend/MoptPaymentEcs.php createrOrUpdateAndForwardUser() function:
```
        if ($this->isUserLoggedIn($session)) {
            $user = $this->updateUserAddresses($payData, $session, $paymentId);
            if ($user === null) {
                return $this->ecsAbortAction();
            }
        }
        $this->createUserWithoutAccount($payData, $session, $paymentId);
```

Was this change made intentionally and if yes, could you explain the reasoning behind it please?

Thanks in advanced.

Best regards
Marco Fritsche

**Comment by fjbender** [comment]
This change was introduced with #124 and makes sure that the address coming from PayPal is always used as invoice and shipping address, because this is required by PayPal.

What's the expected behavior here for logged in users? Especially when the addresses in Shopware and PayPal differ.

**Comment by mfritsche-dotsource** [comment]
Thank you for your reply.

Personally i would expect that the currently existing addresses in the user session will be overwritten with the ones from Paypal and the user stays logged in.

That way the order will stay linked to the user and can be reviewed in his account and used for other services like RMA. This information will be lost and inaccessible if a new user is created and the order treated as fast order.

Best regards
Marco Fritsche

**Comment by fjbender** [comment]
That makes sense to me. However, we chose to go the straightforward way and just treat every PayPal ECS user as a completely new customer to be completely safe in terms of PayPal Buyer/Seller Protection.

As we recently changed this behavior, I will, unfortunately, not have room for this in one of the upcoming weeks. You are very welcome to propose a pull request and we can discuss the changes here with the team.
